packages/fuels-types/src/param_types.rs

```diff
@@ -1295,53 +1334,116 @@ mod tests {
 
-        assert!(!base_vector.contains_nested_vectors());
-        assert!(ParamType::Vector(Box::from(base_vector.clone())).contains_nested_vectors());
+        assert!(!base_vector.contains_nested_heap_types());
+        assert!(ParamType::Vector(Box::from(base_vector.clone())).contains_nested_heap_types());
 
-        assert!(!ParamType::Array(Box::from(ParamType::U8), 10).contains_nested_vectors());
-        assert!(ParamType::Array(Box::from(base_vector), 10).contains_nested_vectors());
+        assert!(!ParamType::Array(Box::from(ParamType::U8), 10).contains_nested_heap_types());
+        assert!(ParamType::Array(Box::from(base_vector), 10).contains_nested_heap_types());
 
-        assert!(!ParamType::Tuple(param_types_no_nested_vec.clone()).contains_nested_vectors());
-        assert!(ParamType::Tuple(param_types_nested_vec.clone()).contains_nested_vectors());
+        assert!(!ParamType::Tuple(param_types_no_nested_vec.clone()).contains_nested_heap_types());
+        assert!(ParamType::Tuple(param_types_nested_vec.clone()).contains_nested_heap_types());
 
         assert!(!ParamType::Struct {
             name: "StructName".to_string(),
             generics: param_types_no_nested_vec.clone(),
             fields: tuples_no_nested_vec.clone(),
         }
-        .contains_nested_vectors());
+        .contains_nested_heap_types());
         assert!(ParamType::Struct {
             name: "StructName".to_string(),
             generics: param_types_nested_vec.clone(),
             fields: tuples_no_nested_vec.clone()
         }
-        .contains_nested_vectors());
+        .contains_nested_heap_types());
         assert!(ParamType::Struct {
             name: "StructName".to_string(),
             generics: param_types_no_nested_vec.clone(),
             fields: tuples_with_nested_vec.clone()
         }
-        .contains_nested_vectors());
+        .contains_nested_heap_types());
 
         assert!(!ParamType::Enum {
             name: "EnumName".to_string(),
             variants: EnumVariants::new(tuples_no_nested_vec.clone())?,
             generics: param_types_no_nested_vec.clone()
         }
-        .contains_nested_vectors());
+        .contains_nested_heap_types());
         assert!(ParamType::Enum {
             name: "EnumName".to_string(),
             variants: EnumVariants::new(tuples_with_nested_vec)?,
             generics: param_types_no_nested_vec
         }
-        .contains_nested_vectors());
+        .contains_nested_heap_types());
         assert!(ParamType::Enum {
             name: "EnumName".to_string(),
             variants: EnumVariants::new(tuples_no_nested_vec)?,
             generics: param_types_nested_vec
         }
-        .contains_nested_vectors());
+        .contains_nested_heap_types());
+        Ok(())
+    }
 
+    #[test]
+    fn contains_no_nested_heap_type_nested_bytes_complex_types() -> Result<()> {
```

Comment:
This naming is confusing. Kind of seems to have two parts. `contains_no_nested_heap_type` and `nested_bytes_complex_types`. But 'complex' types containing `Bytes` **have** nested heap types.